sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1086 @@
+                        == NullnessAnnotation.RETURN_VALUES_ARE_NONNULL_BY_DEFAULT;
+                final boolean isMethodOverridden = isMethodOverridden();
+
+                if (returnValuesAreNonnullByDefault) {
+                    if (!isMethodOverridden) {
+                        checkContainsAny(MSG_REDUNDANT_NONNULL_RETURN_ANNOTATION,
+                                NullnessAnnotation.NONNULL);
+                    }
+                }
+                else {
+                    checkContainsNone(MSG_RETURN_WITHOUT_NULLNESS_ANNOTATION,
+                            NullnessAnnotation.CHECK_FOR_NULL, NullnessAnnotation.NONNULL,
+                            NullnessAnnotation.OVERRIDE);
+                }
+
+                if (isMethodOverridden && !allowOverridingReturnValue) {
+                    checkContainsAny(MSG_OVERRIDDEN_METHODS_ALLOW_ONLY_NONNULL,
+                            NullnessAnnotation.CHECK_FOR_NULL, NullnessAnnotation.NULLABLE);
+                }
+
+                if (isMethodOverridden) {
+                    checkContainsAny(MSG_NEED_TO_INHERIT_PARAM_ANNOTATIONS,
+                            NullnessAnnotation.PARAMETERS_ARE_NONNULL_BY_DEFAULT);
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Class ConstructorJsr305Handler. Check a constructor.
+     */
+    private final class ConstructorJsr305Handler extends AbstractMethodJsr305Handler {
+
+        /**
+         * Constructor.
+         * @param ast
+         *        the ast
+         */
+        private ConstructorJsr305Handler(final DetailAST ast) {
+            super(ast);
+        }
+
+        /**
+         * Check for return type nullness annotations (which are illegal).
+         */
+        @Override
+        protected void runReturnAnnotationCheck() {
+            checkContainsAny(MSG_CONSTRUCTOR_WITH_RETURN_ANNOTATION,
+                    NullnessAnnotation.CHECK_FOR_NULL, NullnessAnnotation.CHECK_RETURN_VALUE,
+                    NullnessAnnotation.NONNULL, NullnessAnnotation.NULLABLE,
+                    NullnessAnnotation.OVERRIDE);
+        }
+
+    }
+
+    /**
+     * An abstract check, the base for checks on parameters, methods, classes. Class
+     * AbstractJsr305Check.
+     */
+    public abstract class AbstractJsr305Handler {
+
+        /** Has a violation been found. */
+        private boolean violationFound;
+        /** The found annotations. */
+        private final Set<NullnessAnnotation> annotations;
+        /** The ast. */
+        private final DetailAST ast;
+
+        /**
+         * Construtor.
+         * @param ast
+         *        the ast
+         */
+        private AbstractJsr305Handler(final DetailAST ast) {
+            this.ast = ast;
+            this.violationFound = false;
+            annotations = findAnnotation();
+        }
+
+        /**
+         * Run the actual check.
+         */
+        public final void check() {
+            runcheck();
```

Comment:
Done.

Reply:
> @mbert I assume you missed the review items listed inside the "Load more..." section.

You're right, I missed them because they were collapsed. Thank you.